sdk/keyvault/keyvault/inc/az_keyvault.h

```diff
@@ -114,7 +94,22 @@ AZ_NODISCARD az_result az_keyvault_keys_key_create(
     az_keyvault_keys_client * client,
     az_span const key_name,
     az_keyvault_json_web_key_type const json_web_key_type,
-    az_keyvault_keys_keys_options const * const options,
+    az_http_response * const response);
+
+/**
+ * @brief Creates a new key, stores it, then returns key parameters and attributes to the client.
+ * The create key operation can be used to create any key type in Azure Key Vault. If the named key
+ * already exists, Azure Key Vault creates a new version of the key. It requires the keys/create
+ * permission.
+ *
+ * @param client
+ * @param options
+ * @param response
+ * @return AZ_NODISCARD az_keyvault_keys_key_create
+ */
+AZ_NODISCARD az_result az_keyvault_keys_key_create_with_options(
```

Comment:
Do we really need a *_with_options functions? Can we just have a az_keyvault_keys_key_create function that always takes an options which can be NULL? This way, we have fewer functions to write, document, & test. Also, it is very C idiomatic to have functions that take NULL for default behavior.

Reply:
I add this other function to have something similar to Java SDK.
We would need to make some more refactoring for having options as null because options are supposed to be created with a name and type, witch won't be the case if name and type are arguments passed to create function.

Usually for testing we only use the las wrapper function. In this case, create(name, type) calles create_with_options(options).  So we just test one to have both.

I like that we can also avoid validating `if(options != NULL)` .

And about documentation, I think it's kind the same ???  one way we need to explain two different behaviors depending on NULL.
Other way, each explanation has its own